.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  build-linux-gcc:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - name: build toolchain
+        run: |
+          make all
+        env:
+          CC: gcc
+          CXX: g++
+      - name: assemble examples
+        run: |
+          make examples
+  build-linux-clang:
```

Comment:
@zhiayang I might be missing something, but i think this one shoud use clang

Reply:
you are right, 5Head.